python/chemiscope/explore.py

```diff
@@ -98,13 +109,33 @@ def soap_kpca_featurize(frames):
     if properties is None:
         properties = {}
 
+    # Pick frames and properties related to the environments if provided
+    if environments is not None:
+        # Sort environments by structure id and atom id
+        environments = sorted(environments, key=lambda x: (x[0], x[1]))
+
+        # Check structure indexes
+        unique_structures = list({env[0] for env in environments})
+        if any(index >= len(frames) for index in unique_structures):
+            raise IndexError(
+                "Some or more indices are greater than the length of the frames"
+            )
+
+        if len(unique_structures) != len(frames):
+            # Pick frames corresponding to the environments
+            frames = [frames[index] for index in unique_structures]
+
+            # Pick properties corresponding to the environments
+            properties = _extract_properties_by_environments(
+                properties, unique_structures
+            )
```

Comment:
I don't understand why this is needed. chemiscope should be fine with extra values in the structure properties, and when staying in 'structure' mode we want to be able to use all of them.

Reply:
Indeed, we don't need it if we provide the environemnts to `chemiscope.show`